app/Http/Controllers/ReportController.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Transaction;
+use App\Models\TransactionCategory;
+use App\Services\UserService;
+use Backpack\CRUD\app\Http\Controllers\CrudController;
+use Backpack\CRUD\app\Library\CrudPanel\CrudPanel;
+use Backpack\CRUD\app\Library\CrudPanel\CrudPanelFacade as CRUD;
+
+/**
+ * Class SubjectCategoryCrudController
+ *
+ * @property-read CrudPanel $crud
+ */
+class ReportController extends CrudController
+{
+    use \Backpack\CRUD\app\Http\Controllers\Operations\CreateOperation;
+    use \Backpack\CRUD\app\Http\Controllers\Operations\DeleteOperation;
+    use \Backpack\CRUD\app\Http\Controllers\Operations\ListOperation;
+    use \Backpack\CRUD\app\Http\Controllers\Operations\UpdateOperation;
+    use \Backpack\Pro\Http\Controllers\Operations\InlineCreateOperation;
+    
+    public function setup(): void
+    {
+        CRUD::setModel(\App\Models\Transaction::class); // Transaction
+        CRUD::setRoute(config('backpack.base.route_prefix').'/report');
+        CRUD::setEntityNameStrings('transaction', 'transactions');
+
+        if (!UserService::hasAccessTo('bookkeeping')) {
+            $this->crud->denyAllAccess();
+        }
+    }
+
+    protected function setupListOperation(): void
+    {
+        CRUD::column('name');
+        CRUD::column('description');
+    }
+
+    protected function setupCreateOperation(): void
+    {
+        CRUD::setValidation(\App\Http\Requests\ReportRequest::class);
+
+        CRUD::field('name');
+        CRUD::field('description');
+    }
+
+    protected function setupUpdateOperation(): void
+    {
+        $this->setupCreateOperation();
+    }
+    
+    public function showReport()
+{
+    $incomeCategories = TransactionCategory::where('type', 'INCOME')->pluck('id')->toArray();
```

Comment:
Use the Enum value, not a hardcoded `"INCOME"` -> `\App\Enums\TransactionTypeEnum::INCOME->value`

Reply:
And this?